### PR TITLE
tls: skip parallel client-cert test under valgrind

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**
This testcase is having a lot of spurious failures, but apparently only
while running under valgrind.

### CODE REVIEW

**Comment by allisonkarlitskaya** [comment]
Bonus commit: add an alarm() so that we don't clog semaphore for too long in case this still blocks under non-valgrind.

File: src/tls/test-server.c [hunk 1]
```diff
@@ -585,6 +589,14 @@ test_tls_client_cert_parallel (TestCase *tc, gconstpointer data)
   pid_t pid;
   int status;
 
+  /* XXX TODO This testcase runs slowly under valgrind and sometimes
```
Comment (Gundersanne) [inline src/tls/test-server.c]: Usually we use `HACK:` for hacks :)

File: src/tls/test-server.c [hunk 2]
```diff
@@ -585,6 +589,14 @@ test_tls_client_cert_parallel (TestCase *tc, gconstpointer data)
...
+   * fails inexplicably.  It's not likely that we're going to find leaks
+   * in the code with this test, so running it under valgrind is just
+   * costing us pain.  Disable it for now.
+   */
+  if (cockpit_test_skip_slow ())
```
Comment (Gundersanne) [inline src/tls/test-server.c]: is there a way to skip this only when running under valgrind?

Comment (allisonkarlitskaya) [reply src/tls/test-server.c]: That's exactly what this does.  The code for `cockpit_test_skip_slow()` looks like:

```
  if (g_getenv ("COCKPIT_SKIP_SLOW_TESTS"))
    {
      g_test_skip ("Skipping slow tests");
      return TRUE;
    }
```

with `COCKPIT_SKIP_SLOW_TESTS=1` set from `Makefile.am` for the `check-memory` rule.

Comment (Gundersanne) [reply src/tls/test-server.c]: oooh ok

**Pull request merged**